src/psyclone/transformations.py

```diff
@@ -2779,13 +2779,15 @@ class Dynamo0p3KernelConstTrans(Transformation):
     # for different orders. Formulas kindly provided by Tom
     # Melvin. See the Qr table at http://femtable.org/background.html,
     # for computed values of w0, w1, w2 and w3 up to order 7.
-    space_to_dofs = {"w3":     (lambda n: (n+1)**3),
-                     "w2":     (lambda n: 3*(n+2)*(n+1)**2),
-                     "w1":     (lambda n: 3*(n+2)**2*(n+1)),
-                     "w0":     (lambda n: (n+2)**3),
-                     "wtheta": (lambda n: (n+2)*(n+1)**2),
-                     "w2h":    (lambda n: 2*(n+2)*(n+1)**2),
-                     "w2v":    (lambda n: (n+2)*(n+1)**2)}
+    space_to_dofs = {"w3":       (lambda n: (n+1)**3),
+                     "w2":       (lambda n: 3*(n+2)*(n+1)**2),
+                     "w1":       (lambda n: 3*(n+2)**2*(n+1)),
+                     "w0":       (lambda n: (n+2)**3),
+                     "wtheta":   (lambda n: (n+2)*(n+1)**2),
+                     "w2h":      (lambda n: 2*(n+2)*(n+1)**2),
+                     "w2v":      (lambda n: (n+2)*(n+1)**2),
+                     "w2broken": (lambda n: 3*(n+1)**2*(n+2)),
```

Comment:
Nice bit of maths :-)

Reply:
If you worked them our rather than asking Tom then you should add your name to the "Formulas kindly provided by" comment.

Reply:
Actually, I believe Thomas Gibson worked these formulas out whilst adding the new spaces into the [function_space_constructor_helper_functions_mod.F90](https://code.metoffice.gov.uk/trac/lfric/browser/LFRic/trunk/infrastructure/source/function_space/function_space_constructor_helper_functions_mod.F90) (where I found them) so I added his name.